### PR TITLE
Allow egress-http-proxy.sh to be run as non-root

### PR DESCRIPTION
**Pull request by yrro**
I'd like to be able to deploy this image without running as root.

The necessary changes are quite simple:

 * Generate squid.conf in a writable location (`/tmp`)  
 * Disable pidfile creation

### CODE REVIEW

**Comment by Miciah** [comment]
Since https://github.com/openshift/release/pull/48854 merged, we now have unit tests.  
/ok-to-test

File: egress/http-proxy/egress-http-proxy.sh [hunk 1]
```diff
@@ -90,4 +91,4 @@ sed -e 's/^/  /' ${CONF}
 echo ""
 echo ""
 
-exec squid -N
+exec squid -N -f /tmp/squid.conf
```
Comment (frobware) [inline egress/http-proxy/egress-http-proxy.sh:94]: We should reference `"${CONF}"`.

Comment (frobware) [reply egress/http-proxy/egress-http-proxy.sh:94]: /lgtm cancel

Let's reference the shell variable -- https://github.com/openshift/images/pull/162#pullrequestreview-1977153552.

Comment (yrro) [reply egress/http-proxy/egress-http-proxy.sh]: Done, and I've changed `${CONF}` to `"${CONF}"` elsewhere in the script.

**Pull request closed without merging**